apis/services/service.go

```diff
@@ -0,0 +1,65 @@
+// +groupName=dscinitialization.opendatahub.io
+package services
+
+import (
+	operatorv1 "github.com/openshift/api/operator/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+)
+
+// ManagementSpec struct defines the component's management configuration.
+// +kubebuilder:object:generate=true
+type ManagementSpec struct {
+	// Set to one of the following values:
+	//
+	// - "Managed" : the operator is actively managing the component and trying to keep it active.
+	//               It will only upgrade the component if it is safe to do so
+	//
+	// - "Removed" : the operator is actively managing the component and will not install it,
+	//               or if it is installed, the operator will try to remove it
+	//
+	// +kubebuilder:validation:Enum=Managed;Removed
+	ManagementState operatorv1.ManagementState `json:"managementState,omitempty"`
+}
+
+// Service struct defines the basis for each OpenDataHub component configuration.
+// +kubebuilder:object:generate=true
+type Service struct {
+	ManagementSpec `json:",inline"`
+}
+
+type ManifestsConfig struct {
+	// uri is the URI point to a git repo with tag/branch. e.g.  https://github.com/org/repo/tarball/<tag/branch>
+	// +optional
+	// +kubebuilder:default:=""
+	// +operator-sdk:csv:customresourcedefinitions:type=spec,order=1
+	URI string `json:"uri,omitempty"`
+
+	// contextDir is the relative path to the folder containing manifests in a repository, default value "manifests"
+	// +optional
+	// +kubebuilder:default:="manifests"
+	// +operator-sdk:csv:customresourcedefinitions:type=spec,order=2
+	ContextDir string `json:"contextDir,omitempty"`
+
+	// sourcePath is the subpath within contextDir where kustomize builds start. Examples include any sub-folder or path: `base`, `overlays/dev`, `default`, `odh` etc.
+	// +optional
+	// +kubebuilder:default:=""
+	// +operator-sdk:csv:customresourcedefinitions:type=spec,order=3
+	SourcePath string `json:"sourcePath,omitempty"`
+}
+
+// +kubebuilder:object:generate=true
+type Status struct {
+	Phase              string             `json:"phase,omitempty"`
+	Conditions         []metav1.Condition `json:"conditions,omitempty"`
+	ObservedGeneration int64              `json:"observedGeneration,omitempty"`
+}
+
+type WithStatus interface {
+	GetStatus() *Status
+}
+
+type ServiceObject interface {
+	client.Object
+	WithStatus
+}
```

Comment:
Not for this PR, but we can probably move the `Status`, `WithStatus`, `ServiceObject` to a shared packages and reuse it across all our at least internal APIs